lms/djangoapps/grades/tests/test_new.py

```diff
@@ -239,56 +240,167 @@ def test_save_and_load(self):
+        self.assertEqual(score.all_total.possible, self.ACTUAL_TOTAL_POSSIBLE)
+
+        # The actual possible score for the six provided blocks.  The number
+        # below is from mocking.
+
+        earned_per_block = 3.0
+        possible_per_block = 7.0
+        with mock_get_score(earned_per_block, possible_per_block):
+            score = subsection_factory.update(self.seq1)
+        self.assertEqual(score.all_total.earned, self.SCORED_BLOCK_COUNT * earned_per_block)
+        self.assertEqual(score.all_total.possible, self.SCORED_BLOCK_COUNT * possible_per_block)
+
+    @ddt.data(
+        u'capa',
+        u'ora',
+        u'lti',
+        u'sga',
+    )
+    def test_loading_different_problem_types(self, problem_type):
+        """
+        Test that transformation works for various block types
+        """
+        # submit one score.
+
+
+@ddt.ddt
+class TestVariedMetadata(ModuleStoreTestCase, ProblemSubmissionTestMixin):
+    """
+    Test that changing the metadata on a block has the desired effect on the
+    persisted score.
+    """
+    default_problem_metadata = {
+        u'graded': True,
+        u'weight': 2.5,
+        u'due': datetime.datetime(2099, 3, 15, 12, 30, 0, tzinfo=pytz.utc),
+    }
+
+    def setUp(self):
+        super(TestVariedMetadata, self).setUp()
+        self.course = CourseFactory.create()
         self.chapter = ItemFactory.create(
             parent=self.course,
-            category=u'chapter',
-            display_name=u'Test Chapter'
+            category="chapter",
+            display_name="Test Chapter"
         )
-        self.seq1 = ItemFactory.create(
+        self.sequence = ItemFactory.create(
             parent=self.chapter,
-            category=u'sequential',
-            display_name=u'Test Sequential 1',
+            category='sequential',
+            display_name="Test Sequential 1",
             graded=True
         )
-        self.vert1 = ItemFactory.create(
-            parent=self.seq1,
-            category=u'vertical',
-            display_name=u'Test Vertical 1'
+        self.vertical = ItemFactory.create(
+            parent=self.sequence,
+            category='vertical',
+            display_name='Test Vertical 1'
         )
+        self.problem_xml = u'''
+            <problem url_name="capa-optionresponse">
+              <optionresponse>
+                <optioninput options="('Correct', 'Incorrect')" correct="Correct"></optioninput>
+                <optioninput options="('Correct', 'Incorrect')" correct="Correct"></optioninput>
+              </optionresponse>
+            </problem>
+        '''
+        self.request = get_request_for_user(UserFactory())
+        self.client.login(username=self.request.user.username, password="test")
+        CourseEnrollment.enroll(self.request.user, self.course.id)
 
-    def _get_fresh_subsection_score(self, course_structure, subsection):
+    def _get_fresh_subsection_score(self, subsection):
```

Comment:
Since you're re-creating the `SubsectionGradeFactory` in each test run, you don't need to call `update`.  `create` should work just fine.  Calling `create` instead may make it more clear to understand the test.  Otherwise, the readers (like @sanfordstudent and myself :) ) may get confused about why an `update` was even required.